src/test/java/com/google/sps/servlets/ProfessorDataTest.java

```diff
@@ -0,0 +1,196 @@
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.data.ProfDataHolder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import javax.servlet.http.HttpServletRequest;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public final class ProfessorDataTest {
+  private static final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  private AddSchoolData schoolData;
+  private ProfessorData professorData;
+
+  @BeforeEach
+  public void setUp() {
+    MockitoAnnotations.initMocks(this);
+    helper.setUp();
+    schoolData = new AddSchoolData();
+    professorData = new ProfessorData();
+  }
+
+  @AfterEach
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Mock HttpServletRequest request;
+  @Mock HttpServletRequest requestB;
+  @Mock HttpServletRequest requestC;
+
+  @Test
+  public void GettingRatingData_AllProperties() throws EntityNotFoundException {
+    DatastoreService db = DatastoreServiceFactory.getDatastoreService();
+
+    List<Long> expectedHoursListA = new ArrayList(Arrays.asList((long) 12));
+    List<Long> expectedDifficultyListA = new ArrayList(Arrays.asList((long) 7));
+    List<Long> expectedPerceptionListA = new ArrayList(Arrays.asList((double) 0.8));
+    List<Long> expectedCommentsListA = new ArrayList(Arrays.asList("Great"));
+
+    List<Long> expectedHoursListB = new ArrayList(Arrays.asList((long) 11));
+    List<Long> expectedDifficultyListB = new ArrayList(Arrays.asList((long) 6));
+    List<Long> expectedPerceptionListB = new ArrayList(Arrays.asList((double) 0.79));
+    List<Long> expectedCommentsListB = new ArrayList(Arrays.asList("Good"));
+
+    createRequest(
+        /* requestServelt */ request,
+        /* schoolName */ "mit",
+        /* courseName */ "6.006",
+        /* termName */ "Spring 2020",
+        /* units */ "12",
+        /* profName */ "Jason Ku");
+    createRequest(
+        /* requestServelt */ requestB,
+        /* schoolName */ "mit",
+        /* courseName */ "6.006",
+        /* termName */ "Fall 2020",
+        /* units */ "12",
+        /* profName */ "Jason Ku");
+
+    schoolData.addSchoolData(db, request);
+    schoolData.addSchoolData(db, requestB);
+    Key parent = findQueryMatch(db, "Term", "term", "Spring 2020").get(0).getKey();
```

Comment:
Just a thought, this entity (or at least the key) feels rather like something that "addSchoolData" should be returning

Reply:
I will look into this. Let me get all the professor related PR's in then I will go back and refactor this

Reply:
Good to add a TODO comment in that case so you can track this